### PR TITLE
Separate manifold_id from boundary_id

### PR DESCRIPTION
**Pull request by luca-heltai**
This is alternative to #5932. 

A lot of tests will be broken with this commit, and a lot of changes need to be addressed also in the documentation. 

Before we go forward with this, we should discuss if this is the right way to go.

If it is, I'm willing to fix all tests (and @drwells offered to help as well... ;) ...).

### CODE REVIEW

**Comment by luca-heltai** [comment]
If we want only part of this, then let's argue about what should be fixe and what should be postponed. 
For example, we may want to set manifold ids, but not attach the manifold description (for whatever reason).

**Comment by luca-heltai** [comment]
These are the currently failing tests:

```
96% tests passed, 129 tests failed out of 3549

Total Test time (real) = 4382.63 sec

The following tests FAILED:

	  4 - aniso/refine_and_coarsen_anisotropic.debug (Failed)
	  8 - aniso/mesh_3d_21.debug (Failed)
	 11 - numerics/create_laplace_matrix_04.debug (Failed)
	 13 - numerics/create_laplace_matrix_02b.debug (Failed)
	 14 - numerics/no_flux_09.debug (Failed)
	 17 - numerics/vectors_rhs_03.debug (Failed)
	 18 - numerics/derivative_approximation_01.debug (Failed)
	 19 - numerics/create_mass_matrix_01.debug (Failed)
	 20 - numerics/create_laplace_matrix_02.debug (Failed)
	 24 - numerics/no_flux_14.debug (Failed)
	 28 - numerics/vectors_boundary_rhs_02.debug (Failed)
	 29 - numerics/create_mass_matrix_05.debug (Failed)
	 30 - numerics/create_mass_matrix_03.debug (Failed)
	 40 - numerics/no_flux_08.debug (Failed)
	 41 - numerics/create_mass_matrix_04.debug (Failed)
	 44 - numerics/no_flux_10.debug (Failed)
	 46 - numerics/create_mass_matrix_04b.debug (Failed)
	 49 - numerics/create_laplace_matrix_01b.debug (Failed)
	 51 - numerics/create_laplace_matrix_01.debug (Failed)
	 59 - numerics/derivatives.debug (Failed)
	 62 - numerics/vectors_boundary_rhs_03.debug (Failed)
	 63 - numerics/project_dgp_monomial_03.debug (Failed)
	 64 - numerics/project_nedelec_03.debug (Failed)
	 68 - numerics/create_laplace_matrix_03b.debug (Failed)
	 72 - numerics/error_estimator.debug (Failed)
	 82 - numerics/create_mass_matrix_02.debug (Failed)
	 84 - numerics/project_q_system_03.debug (Failed)
	107 - numerics/create_mass_matrix_02b.debug (Failed)
	108 - numerics/create_laplace_matrix_03.debug (Failed)
	112 - numerics/project_q_03.debug (Failed)
	113 - numerics/no_flux_12.debug (Failed)
	114 - numerics/project_boundary_01.debug (Failed)
	116 - numerics/vectors_boundary_rhs_01.debug (Failed)
	122 - numerics/project_dgq_03.debug (Failed)
	125 - numerics/create_mass_matrix_01b.debug (Failed)
	128 - numerics/project_dgp_03.debug (Failed)
	129 - numerics/create_mass_matrix_03b.debug (Failed)
	133 - numerics/create_laplace_matrix_04b.debug (Failed)
	143 - numerics/vectors_point_source_01.debug (Failed)
	144 - numerics/vectors_rhs_01.debug (Failed)
	178 - mappings/mapping.debug (Failed)
	182 - mappings/mapping_manifold_04.debug (Failed)
	199 - mappings/mapping_manifold_05.debug (Failed)
	213 - mappings/mapping_c1.debug (Failed)
	242 - dofs/dof_renumbering.debug (Failed)
	480 - lac/matrices_02.debug (Failed)
	532 - lac/inhomogeneous_constraints_nonsymmetric.debug (Failed)
	558 - lac/filtered_matrix_06.debug (Failed)
	688 - manifold/transfinite_manifold_02.debug (Failed)
	711 - manifold/polar_manifold_02.debug (Failed)
	1074 - hp/create_laplace_matrix_04.debug (Failed)
	1075 - hp/create_laplace_matrix_02b.debug (Failed)
	1079 - hp/vectors_rhs_03.debug (Failed)
	1087 - hp/create_mass_matrix_01.debug (Failed)
	1093 - hp/create_laplace_matrix_02.debug (Failed)
	1097 - hp/vectors_rhs_hp_01.debug (Failed)
	1104 - hp/boundary_matrices.debug (Failed)
	1107 - hp/vectors_boundary_rhs_hp_03.debug (Failed)
	1109 - hp/vectors_boundary_rhs_02.debug (Failed)
	1113 - hp/create_mass_matrix_05.debug (Failed)
	1120 - hp/create_mass_matrix_03.debug (Failed)
	1125 - hp/matrices_hp.debug (Failed)
	1136 - hp/create_mass_matrix_04.debug (Failed)
	1139 - hp/vectors_boundary_rhs_hp_01.debug (Failed)
	1149 - hp/create_mass_matrix_04b.debug (Failed)
	1152 - hp/matrices.debug (Failed)
	1156 - hp/create_laplace_matrix_01b.debug (Failed)
	1159 - hp/create_laplace_matrix_01.debug (Failed)
	1169 - hp/vectors_rhs_hp_03.debug (Failed)
	1177 - hp/vectors_boundary_rhs_03.debug (Failed)
	1194 - hp/create_laplace_matrix_03b.debug (Failed)
	1204 - hp/step-5.debug (Failed)
	1222 - hp/create_mass_matrix_02.debug (Failed)
	1235 - hp/step-2.debug (Failed)
	1255 - hp/vectors_rhs_02.debug (Failed)
	1259 - hp/create_mass_matrix_02b.debug (Failed)
	1261 - hp/create_laplace_matrix_03.debug (Failed)
	1270 - hp/boundary_matrices_hp.debug (Failed)
	1280 - hp/vectors_boundary_rhs_01.debug (Failed)
	1286 - hp/vectors_point_source_hp_01.debug (Failed)
	1292 - hp/create_mass_matrix_01b.debug (Failed)
	1299 - hp/create_mass_matrix_03b.debug (Failed)
	1304 - hp/create_laplace_matrix_04b.debug (Failed)
	1312 - hp/vectors_point_source_01.debug (Failed)
	1313 - hp/vectors_rhs_01.debug (Failed)
	1498 - codim_one/extract_boundary_mesh_01.debug (Failed)
	1507 - codim_one/boundary_indicator_01.debug (Failed)
	1523 - codim_one/extract_boundary_mesh_10.debug (Failed)
	1526 - codim_one/grid_refinement.debug (Failed)
	1546 - codim_one/extract_boundary_mesh_03.debug (Failed)
	1558 - codim_one/bem.debug (Failed)
	1736 - vector_tools/boundaries_complex_hp.debug (Failed)
	1741 - vector_tools/boundaries.debug (Failed)
	1742 - vector_tools/boundaries_complex.debug (Failed)
	1746 - grid/mesh_3d_14.debug (Failed)
	1766 - grid/mesh_3d_12.debug (Failed)
	1815 - grid/maximal_cell_diameter.debug (Failed)
	1826 - grid/minimal_cell_diameter.debug (Failed)
	1832 - grid/grid_hyper_shell.debug (Failed)
	1843 - grid/mesh_3d_6.debug (Failed)
	1857 - grid/grid_transform.debug (Failed)
	1865 - grid/mesh_3d_7.debug (Failed)
	1867 - grid/recursively_set_material_id.debug (Failed)
	1870 - grid/grid_output_input.debug (Failed)
	1885 - grid/mesh_3d_26.debug (Failed)
	1897 - grid/grid_out_svg_02.debug (Failed)
	1926 - grid/grid_transform_coefficient.debug (Failed)
	1936 - grid/manifold_ids_on_boundary_01.debug (Failed)
	1967 - grid/grid_out_svg_01.debug (Failed)
	2359 - fe/cell_similarity_08.debug (Failed)
	2364 - fe/cell_similarity_dgp_nonparametric_05.debug (Failed)
	2402 - fe/cell_similarity_dgp_nonparametric_07.debug (Failed)
	2436 - fe/cell_similarity_05.debug (Failed)
	2455 - fe/cell_similarity_dgp_nonparametric_06.debug (Failed)
	2485 - fe/cell_similarity_dgp_monomial_06.debug (Failed)
	2492 - fe/cell_similarity_dgp_nonparametric_08.debug (Failed)
	2493 - fe/cell_similarity_dgp_monomial_05.debug (Failed)
	2513 - fe/cell_similarity_dgp_monomial_08.debug (Failed)
	2514 - fe/cell_similarity_11.debug (Failed)
	2661 - fe/mass_matrix_arbitrary_nodes.debug (Failed)
	2667 - fe/cell_similarity_dgp_monomial_07.debug (Failed)
	2670 - fe/cell_similarity_06.debug (Failed)
	2685 - fe/cell_similarity_07.debug (Failed)
	2761 - bits/face_orientation_and_fe_nedelec_01.debug (Failed)
	2789 - bits/find_cell_alt_3.debug (Failed)
	2841 - bits/step-5.debug (Failed)
	2868 - bits/face_orientation_and_fe_q_01.debug (Failed)
	2896 - bits/step-2.debug (Failed)
	2905 - bits/find_cell_3.debug (Failed)
```

**Comment by luca-heltai** [comment]
@dealii/dealii  , Before I keep going with the other failing tests, I need feedback. 

This PR breaks backward compatibility (in my opinion in a good way), by associating the right manifold descriptor to all meshes we generate internally. 

In many tests, we used to create hyperballs and refine them without attaching a manifold. All of these are now failing, since at refine the new meshes are "smooth", while the old meshes are not (100% of the tests I fixed so far were because of this).

Alternatively, we could add a flag to each function in GridGenerator, defaulting to false:
```
GridGenerator::my_cool_mesh(...., bool attach_manifolds = false);
```

This saves on many failing tests, does not break backward compatibility 99% of the times (since we still set the manifold_id to a value !=  flat_manifold_id on the curved buondary) but does not look at the future... the "right" choice should be, IMO, attach the manifold by default.

What do other think?

**Comment by masterleinad** [comment]
> Alternatively, we could add a flag to each function in GridGenerator, defaulting to false:
> 
> GridGenerator::my_cool_mesh(...., bool attach_manifolds = false);
> 
> This saves on many failing tests, does not break backward compatibility 99% of the times (since we still set the manifold_id to a value != flat_manifold_id on the curved buondary) but does not look at the future... the "right" choice should be, IMO, attach the manifold by default.
> 
> What do other think?

I could think of
```
GridGenerator::my_cool_mesh(...., bool attach_manifolds = true);
```
to be useful. Then, we still have the possibility to request the old behavior.

**Comment by luca-heltai** [comment]
@bangerth , this is essentially what I'm doing in the tests that I already corrected. I do this by calling `set_manifold(0)` (unfortunate name, I must say, since it actually *resets* the manifold)

**Comment by tjhei** [comment]
> Alternatively, we could add a flag to each function in GridGenerator, defaulting to false:

Personally, I don't like bool flags, especially with default values.

> I think the idea of always attaching the right manifold objects is good

Is there always a "right manifold", though? Especially for shells, there seem to exist different options with different pros/cons.

**Comment by jppelteret** [comment]
> I think the idea of always attaching the right manifold objects is good. It is clearly where we ought to be going for the future.

My biggest concern here is that if you transform the grid after a manifold is attached, (e.g. translate it) then to the best of my knowledge the manifold data is not updated as well.

**Comment by bangerth** [comment]
On 04/13/2018 09:34 AM, Luca Heltai wrote:
> (unfortunate name, I must say, since it actually /resets/ the manifold)

Indeed -- now would be the time to change this!

**Comment by bangerth** [comment]
On 04/13/2018 09:43 AM, Timo Heister wrote:
> Is there always a "right manifold", though? Especially for shells, there 
> seem to exist different options with different pros/cons.

I think there is always a manifold that is at least "better" than no 
manifold at all. That doesn't imply that users don't later overrule this 
by adding an "even better" (or simply "different") one.


 > My biggest concern here is that if you transform the grid after a
 > manifold is attached, (e.g. translate it) then to the best of my
 > knowledge the manifold data is not updated as well.

Ah, that's a good point, though :-(

**Comment by tjhei** [comment]
> My biggest concern here is that if you transform the grid after a manifold is attached, (e.g. translate it) then to the best of my knowledge the manifold data is not updated as well.

A ``GridTransfom::translate`` could replace all manifolds with a ``CompositionManifold`` of the old + a translation. Not sure if that goes too far, though.

**Comment by jppelteret** [comment]
> A GridTransfom::translate could replace all manifolds with a CompositionManifold of the old + a translation. Not sure if that goes too far, though.

I agree with this, but translation was one simple example. We'd have to make sure that you could do any combination of transformations with all of the options provided in `GridTools` (e.g. translate -> rotate -> translate) which is a lot to ask for this upcoming milestone.

**Comment by luca-heltai** [comment]
@jppelteret , I agree about the transformation, but this can be done in a later pull request. Right now you have the exact same problem, so this PR is not making things worse. You have to attach a manifold to the triangulation after you have transformed it anyway now, and by doing so, you simply replace the one that is by default attached to the triangulation.

The general solution is also doable with ease. `CompositionManifold` right now works only for chart manifolds, but there is no reasons why one should not be able to make it work with a ChartManifold (based on the transformation) and a standard manifold. I'm not sure if you'll need the inverse transformation, though... I have to think about this...

**Comment by masterleinad** [comment]
> @jppelteret , I agree about the transformation, but this can be done in a later pull request. Right now you have the exact same problem, so this PR is not making things worse. You have to attach a manifold to the triangulation after you have transformed it anyway now, and by doing so, you simply replace the one that is by default attached to the triangulation.

I agree. Let's not make things more complicated here.

**Comment by luca-heltai** [comment]
This is the current situation for my tester, with all libraries on:

```
98% tests passed, 118 tests failed out of 4903

Total Test time (real) = 11194.08 sec

The following tests FAILED:
```
- [ ] mappings/mapping.debug
- [ ] mappings/mapping_manifold_05.debug
- [ ] mappings/mapping_c1.debug
- [ ] zoltan/zoltan_02.mpirun=2.debug
- [ ] zoltan/tria_zoltan_01.mpirun=2.debug   
- [ ] zoltan/tria_zoltan_01.debug
- [ ] zoltan/zoltan_01.mpirun=2.debug
- [ ] grid/grid_transform.debug
- [ ] grid/manifold_ids_on_boundary_01.debug
- [ ] grid/mesh_3d_14.debug
- [ ] grid/grid_out_svg_01.debug
- [ ] grid/minimal_cell_diameter.debug
- [ ] grid/recursively_set_material_id.debug
- [ ] grid/maximal_cell_diameter_mpi.mpirun=4.debug
- [ ] grid/grid_transform_coefficient.debug 
- [ ] grid/mesh_3d_12.debug
- [ ] grid/mesh_3d_26.debug
- [ ] grid/mesh_3d_6.debug
- [ ] grid/grid_output_input.debug
- [ ] grid/mesh_3d_7.debug
- [ ] grid/maximal_cell_diameter.debug
- [ ] grid/grid_out_svg_02.debug
- [ ] grid/grid_hyper_shell.debug
- [ ] grid/minimal_cell_diameter_mpi.mpirun=4.debug
- [ ] codim_one/grid_refinement.debug
- [ ] codim_one/extract_boundary_mesh_03.debug
- [ ] codim_one/extract_boundary_mesh_10.debug
- [ ] codim_one/extract_boundary_mesh_01.debug
- [ ] codim_one/bem.debug
- [ ] codim_one/boundary_indicator_01.debug 
- [ ] hp/step-5.debug
- [ ] hp/create_laplace_matrix_04b.debug
- [ ] hp/create_mass_matrix_05.debug
- [ ] hp/create_laplace_matrix_02.debug
- [ ] hp/create_mass_matrix_02.debug
- [ ] hp/vectors_boundary_rhs_02.debug
- [ ] hp/vectors_point_source_01.debug
- [ ] hp/boundary_matrices.debug
- [ ] hp/vectors_boundary_rhs_03.debug
- [ ] hp/create_mass_matrix_04.debug
- [ ] hp/vectors_point_source_hp_01.debug   
- [ ] hp/vectors_rhs_03.debug
- [ ] hp/create_mass_matrix_03.debug
- [ ] hp/step-2.debug
- [ ] hp/create_laplace_matrix_01b.debug
- [ ] hp/matrices_hp.debug
- [ ] hp/matrices.debug
- [ ] hp/create_laplace_matrix_02b.debug
- [ ] hp/create_mass_matrix_01.debug
- [ ] hp/create_mass_matrix_04b.debug
- [ ] hp/boundary_matrices_hp.debug
- [ ] hp/create_mass_matrix_02b.debug
- [ ] hp/create_laplace_matrix_04.debug
- [ ] hp/vectors_rhs_01.debug
- [ ] hp/create_laplace_matrix_03.debug
- [ ] hp/vectors_rhs_hp_01.debug
- [ ] hp/vectors_rhs_02.debug
- [ ] hp/vectors_rhs_hp_03.debug
- [ ] hp/create_mass_matrix_01b.debug
- [ ] hp/create_mass_matrix_03b.debug
- [ ] hp/vectors_boundary_rhs_hp_01.debug   
- [ ] hp/vectors_boundary_rhs_hp_03.debug   
- [ ] hp/create_laplace_matrix_01.debug
- [ ] hp/create_laplace_matrix_03b.debug
- [ ] hp/vectors_boundary_rhs_01.debug
- [ ] bits/step-5.debug
- [ ] bits/face_orientation_and_fe_nedelec_01.debug
- [ ] bits/step-2.debug
- [ ] bits/find_cell_3.debug
- [ ] bits/face_orientation_and_fe_q_01.debug
- [ ] bits/find_cell_alt_3.debug
- [ ] dofs/dof_renumbering.debug
- [ ] mpi/no_flux_constraints_02.mpirun=1.debug
- [ ] mpi/no_flux_constraints_02.mpirun=9.debug
- [ ] mpi/p4est_3d_constraintmatrix_03.mpirun=10.debug
- [ ] mpi/p4est_2d_constraintmatrix_03.mpirun=4.debug
- [ ] mpi/map_dofs_to_support_points.mpirun=10.debug
- [ ] mpi/p4est_3d_refine_04.mpirun=4.debug 
- [ ] mpi/periodicity_02.mpirun=5.debug
- [ ] mpi/step-37.mpirun=1.debug
- [ ] mpi/periodicity_02.mpirun=7.debug
- [ ] mpi/no_flux_constraints.mpirun=9.debug
- [ ] mpi/p4est_3d_constraintmatrix_03.mpirun=4.debug
- [ ] mpi/map_dofs_to_support_points.mpirun=4.debug
- [ ] mpi/p4est_3d_refine_04.mpirun=10.debug
- [ ] mpi/p4est_3d_refine_04.mpirun=24.debug
- [ ] mpi/p4est_2d_constraintmatrix_03.mpirun=10.debug
- [ ] mpi/periodicity_02.mpirun=3.debug
- [ ] mpi/step-37.mpirun=7.debug
- [ ] mpi/normal_flux_constraints.mpirun=9.debug
- [ ] mpi/p4est_3d_refine_04.mpirun=1.debug 
- [ ] lac/filtered_matrix_06.debug
- [ ] lac/matrices_02.debug
- [ ] lac/inhomogeneous_constraints_nonsymmetric.debug
- [ ] manifold/polar_manifold_02.debug
- [ ] manifold/transfinite_manifold_02.debug
- [ ] distributed_grids/3d_coarse_grid_02.debug
- [ ] numerics/project_parallel_q_03.mpirun=4.debug
- [ ] fe/mass_matrix_arbitrary_nodes.debug  
- [ ] fe/cell_similarity_06.debug
- [ ] fe/fe_enriched_step-36b.debug
- [ ] fe/cell_similarity_11.debug
- [ ] fe/cell_similarity_dgp_nonparametric_07.debug
- [ ] fe/fe_enriched_step-36.debug
- [ ] fe/cell_similarity_dgp_monomial_08.debug
- [ ] fe/cell_similarity_dgp_nonparametric_08.debug
- [ ] fe/cell_similarity_dgp_monomial_05.debug
- [ ] fe/cell_similarity_08.debug
- [ ] fe/cell_similarity_dgp_monomial_07.debug
- [ ] fe/cell_similarity_dgp_monomial_06.debug
- [ ] fe/cell_similarity_05.debug
- [ ] fe/cell_similarity_dgp_nonparametric_06.debug
- [ ] fe/cell_similarity_07.debug
- [ ] fe/cell_similarity_dgp_nonparametric_05.debug
- [ ] vector_tools/boundaries_complex_hp.debug
- [ ] vector_tools/boundaries.debug
- [ ] vector_tools/boundaries_complex.debug

**Comment by luca-heltai** [comment]
Meta #4921

**Comment by jppelteret** [comment]
@luca-heltai @masterleinad You misinterpreted the "message" of my comment. I was only trying to point out that there would be further challenges that needed to be tackled once this mechanism is adopted, and prompt that this deficiency should be noted in the documentation should a fix not be in the pipeline for the next release. Moreover, until now I wasn't aware that we had any generic way to deal with transformations of manifolds. So if I didn't make my concern heard here then where was I to do so? I apologise for not being more explicit that I wanted to discuss these points rather than have an expectation of an immediate fix. 

For the record, I'm completely in favour of it of what's being done here. Thank you for making these changes.

> The general solution is also doable with ease. CompositionManifold right now works only for chart manifolds, but there is no reasons why one should not be able to make it work with a ChartManifold (based on the transformation) and a standard manifold. I'm not sure if you'll need the inverse transformation, though... I have to think about this...

Thanks for the explanation. I'm glad that that this should be extensible to deal with the problem at hand. I'm really excited as the prospect of having a `CompositionManifold` for general manifolds.

**Comment by luca-heltai** [comment]
> @luca-heltai @masterleinad You misinterpreted the "message" of my comment. I was only trying to point out that there would be further challenges that needed to be tackled once this mechanism is adopted, and prompt that this deficiency should be noted in the documentation should a fix not be in the pipeline for the next release. 

@jppelteret , don't apologize! I agree with everything you said in your comment (and we also discussed about this personally). I needed feedback to decide if I had to spend a few hours in doing boring changes to tests... before doing this, I want to make sure we all agree that this is the right direction. Your concerns are perfectly valid ones, and we should open an issue if this gets merged to address them.

> Moreover, until now I wasn't aware that we had any generic way to deal with transformations of manifolds. So if I didn't make my concern heard here then where was I to do so? I apologise for not being more explicit that I wanted to discuss these points rather than have an expectation of an immediate fix.

The solution is there, but it is not generic at the moment. Another issue to open. ;) 

> For the record, I'm completely in favour of it of what's being done here. Thank you for making these changes.

👍

**Comment by masterleinad** [comment]
> @luca-heltai +1
> 
>     I needed feedback to decide if I had to spend a few hours in doing boring changes to tests... before doing this, I want to make sure we all agree that this is the right direction.
> 
> Whichever road we go down to fix the issue, I'm also willing to help fix the tests. There are a lot of them to cover but if we divide the work then it'll go much quicker!

Me too.

**Comment by drwells** [comment]
Quick question: in `hyper_ball` we set the manifold ids on the boundary cells equal to 1. Is there any reason that we cannot also set these equal to zero so that they have the same manifold ids as the boundary?

**Comment by masterleinad** [comment]
I fixed hp, lac and fe so far.

**Comment by masterleinad** [comment]
I also took the freedom to rebase this branch so we can use `reset_manifold`.

**Comment by luca-heltai** [comment]
Ok.

**Comment by jppelteret** [comment]
I'm doing `codim_one` now...

**Comment by luca-heltai** [comment]
Story for not doing much today, but I'm traveling. 

I'll get back to the grid tests tonight.

**Comment by jppelteret** [comment]
I tried to fix `manifold/transfinite_manifold_02.debug` but didn't have much luck. The issue is with `test_cylinder`. Maybe I'll come back to this later today.

**Comment by jppelteret** [comment]
Before I push to the branch again, here's the latest list of failing tests from the tester:

```
99% tests passed, 22 tests failed out of 3550

Total Test time (real) = 4357.09 sec

The following tests FAILED:
	178 - mappings/mapping.debug (Failed)
	182 - mappings/mapping_manifold_04.debug (Failed)
	199 - mappings/mapping_manifold_05.debug (Failed)
	213 - mappings/mapping_c1.debug (Failed)
	242 - dofs/dof_renumbering.debug (Failed)
	688 - manifold/transfinite_manifold_02.debug (Failed)
	1526 - codim_one/grid_refinement.debug (Failed)
	1746 - grid/mesh_3d_14.debug (Failed)
	1766 - grid/mesh_3d_12.debug (Failed)
	1815 - grid/maximal_cell_diameter.debug (Failed)
	1826 - grid/minimal_cell_diameter.debug (Failed)
	1832 - grid/grid_hyper_shell.debug (Failed)
	1843 - grid/mesh_3d_6.debug (Failed)
	1857 - grid/grid_transform.debug (Failed)
	1865 - grid/mesh_3d_7.debug (Failed)
	1867 - grid/recursively_set_material_id.debug (Failed)
	1870 - grid/grid_output_input.debug (Failed)
	1885 - grid/mesh_3d_26.debug (Failed)
	1897 - grid/grid_out_svg_02.debug (Failed)
	1926 - grid/grid_transform_coefficient.debug (Failed)
	1936 - grid/manifold_ids_on_boundary_01.debug (Failed)
	1967 - grid/grid_out_svg_01.debug (Failed)
Errors while running CTest
```

**Comment by drwells** [comment]
It looks like @masterleinad beat me to the tests in `bits/`, but I put up related changes as a PR.

**Comment by masterleinad** [comment]
For me, only
```
manifold/transfinite_manifold_02.debug
mpi/periodicity_02.mpirun=3.debug
mpi/periodicity_02.mpirun=5.debug
mpi/periodicity_02.mpirun=7.debug
```
are still failing.

**Comment by masterleinad** [comment]
For me, all the tests are passing now.

**Comment by luca-heltai** [comment]
This was truly a collective efforts! Thanks @masterleinad , @drwells and @jppelteret !

**Comment by luca-heltai** [comment]
I think we can make changes to the documentation in a followup PR. This is already huge as it is right now.

**Comment by masterleinad** [comment]
We should probably squash this a bit before merging.

**Comment by drwells** [comment]
Thanks.

@luca-heltai my PR against this branch is still open. Can I just push the changes directly onto this branch? I am not sure if I have write access.

**Comment by masterleinad** [comment]
> Can I just push the changes directly onto this branch? I am not sure if I have write access.

Yes, you should have write access by default for all the PRs.

**Comment by drwells** [comment]
This PR is getting monstrously large. I have a few small cleanups but it might be better (particularly now that all the tests pass) if I handle them with follow-up PRs individually so that we can get this done. Thoughts?

**Comment by drwells** [comment]
I propose that we open issues for 'review all documentation related to GridGenerator' and 'update all tutorials using GridGenerator' (indeed, I already opened #6184) and go with this once all tests pass.

**Comment by drwells** [comment]
I believe that I fixed the failing tests. I simply disabled manifold descriptions on the interior so that we matched the old output.

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -54,7 +61,7 @@ namespace GridGenerator
    * tensor product interval $[left,right]^{\text{dim}}$ in the present number
    * of dimensions, where the limits are given as arguments. They default to
    * zero and unity, then producing the unit hypercube.
-   *
+               *
```
Comment (kronbichler) [inline include/deal.II/grid/grid_generator.h]: Sorry to be picky on such a big PR but I assume you want to undo this change.

**Comment by drwells** [comment]
I could go either way on the breaking change of attaching default manifolds: I am under the impression that most people support this decision.

**Comment by luca-heltai** [comment]
It's not going to be silent, but I don't want to keep adding to this PR (which is already huge). I think we should merge, and start right away updating the documentation and the release paper.

**Comment by tamiko** [comment]
Let me be the one making the testsuite unhappy :-)

**Comment by tjhei** [comment]
> @tjhei Any final comments why we should wait?

This has nothing to with waiting or not. It is more of a fundamental question: how often is the default manifold situation not what the user wants and how much code are we breaking by assuming a default like this? I think we need to adjust ASPECT too, right @bangerth ?

The alternative would be to not do anything by default...

**Comment by luca-heltai** [comment]
> @tjhei Any final comments why we should wait?
> 
> This has nothing to with waiting or not. It is more of a fundamental question: how often is the default manifold situation not what the user wants and how much code are we breaking by assuming a default like this? I think we need to adjust ASPECT too, right @bangerth ?
> 
> The alternative would be to not do anything by default...
> 

I'd say 99% of the time, the default is OK.

 You said you didn't want boolean flags with default values, and then didn't object when the decision became to have default manifolds. 4 people spent a lot of time on the outcome  of that discussion. What 's exactly your suggestion now?

**Comment by tjhei** [comment]
> What 's exactly your suggestion now?

Sorry, I did not mean to imply that I object to this PR. I don't have a better suggestion.

**Comment by bangerth** [comment]
On 04/19/2018 11:11 AM, Timo Heister wrote:
> This has nothing to with waiting or not. It is more of a fundamental 
> question: how often is the default manifold situation not what the user 
> wants and how much code are we breaking by assuming a default like this? 
> I think we need to adjust ASPECT too, right @bangerth 

Probably, though I don't think that this will require major adjustments 
-- we probably *already* attach some kind of manifold.

**Comment by kronbichler** [comment]
As nobody seems to have a further suggestion - except for the fact that we have to be careful due to assigning a default manifold, I'll be the one to pull the trigger now.
